lib/engine/game/base.rb

```diff
@@ -1179,16 +1183,21 @@ def discard_discount(train, price)
         (price * (100.0 - self.class::DISCARDED_TRAIN_DISCOUNT.to_f) / 100.0).ceil.to_i
       end
 
-      def end_game!
+      def end_game!(player_initiated: false)
         return if @finished
 
         @finished = true
+        @ended_by_player = player_initiated
         store_player_info
         @round_counter += 1
-        scores = result.map { |name, value| "#{name} (#{format_currency(value)})" }
+        scores = result.map { |id, value| "#{@players.find { |p| p.id == id.to_i }&.name} (#{format_currency(value)})" }
         @log << "-- Game over: #{scores.join(', ')} --"
       end
 
+      def manually_ended?
```

Comment:
probably not necessary and you can just use a attr_reader

Reply:
So just make an attr_reader than a method?

Reply:
yea